src/ui/controllercollection.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Manages UI Controllers.
+ *
+ * @class ControllerCollection
+ * @extends Collection
+ */
+CKEDITOR.define( [
+	'collection',
+	'ckeditorerror'
+], ( Collection, CKEditorError ) => {
+	class ControllerCollection extends Collection {
```

Comment:
Actually, I already have an idea for a customised behaviour that this class should have. I think that `Controller.addChild()` and `Controller.removeChild()` should be moved to `ControllerCollection.add()` and `ControllerCollection.remove()`. Then we would have this:

```
this.children.get( 'buttons' ).add( someButton )
    .then( ... ); // yep, add() would remove a promise as `Controller.addChild() does now
```